intelmq/lib/datatypes.py

```diff
@@ -54,8 +58,85 @@ def toJson(self):
 
 
 class LogLevel(Enum):
-    DEBUG    = 0
-    INFO     = 1
-    WARNING  = 2
-    ERROR    = 3
+    DEBUG = 0
+    INFO = 1
+    WARNING = 2
+    ERROR = 3
     CRITICAL = 4
+
+
+class TimeFormat(str):
+    """
+    Pydantic style Field Type class for bot parameter time_format. Used for validation.
+    """
+
+    def __new__(cls, value: Optional[str] = None):
+        value = "fuzzy" if (value is None or value == "") else value
+        return super().__new__(cls, value)
+
+    def __init__(self, value: Optional[str] = None):
+
+        self.convert: Callable
+        self.format_string: Optional[str] = None
+
+        super(TimeFormat, self).__init__()
+
+        if isinstance(value, TimeFormat):
```

Comment:
Well the `TimeFormat` _is_ `str` as well.

This is to allow code such as this (which is generally fairly common pattern):
```python
t1 = TimeFormat("from_format|%Y-%m-%d %H:%M:%S")
t2 = TimeFormat(t1)
```

By recognizing that the value is `TimeFormat` and not just `str` we can save ourselves some computation because the validation doesn't need to be called again.

I understand the annotations only as recommendations for other code on what should be passed as an argument. Because in the end Python lets you pass any data type even if it contradicts the annotations. So person can't fully count on the annotations being strictly followed. Expecting the above mentioned pattern in some way seems reasonable to me.

Reply:
Explanation added.